### PR TITLE
NEXTEUROPA-12532: Remove HTML5 Warnings during the loadHTML.

### PR DESCRIPTION
**Pull request by voidtek**
https://webgate.ec.europa.eu/CITnet/jira/browse/NEXTEUROPA-12532

### CODE REVIEW

File: profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module [hunk 1]
```diff
@@ -1657,3 +1656,54 @@ function multisite_drupal_toolbox_features_post_restore($op, $items) {
...
+}
+
+/**
+ * Remove PHP Warning logs from HTML5 during a loadHTMLFile.
+ *
+ * Extend SimpleXMLElement function.
+ *
+ * @see $dom->loadHTMLFile;
+ */
+function multisite_drupal_toolbox_load_html_file(&$dom, $htmlfile) {
```
Comment (cyberwolf) [inline profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module]: You shouldn't pass $dom by reference here. An object is always passed by its object identifier and thus the reference is not needed. See http://php.net/manual/en/language.oop5.references.php

File: profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module [hunk 2]
```diff
@@ -1657,3 +1656,54 @@ function multisite_drupal_toolbox_features_post_restore($op, $items) {
...
 }
+
+/**
+ * Remove PHP Warning logs from HTML5 during a loadHTML.
+ *
+ * Extend SimpleXMLElement function.
+ *
+ * @see $dom->loadHTML;
+ */
+function multisite_drupal_toolbox_load_html(&$dom, $html) {
```
Comment (cyberwolf) [inline profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module]: You shouldn't pass $dom by reference here. An object is always passed by its object identifier and thus the reference is not needed. See http://php.net/manual/en/language.oop5.references.php

File: tests/features/tmgmt/tmgmt_poetry.feature [hunk 3]
```diff
@@ -240,3 +240,34 @@ Feature: TMGMT Poetry features
       | Ampersand &, < and > | 'Title contains characters with a special meaning in HTML.'                          |
       | Entities in body     | 'Some text with &amp;, &lt; and &gt;. And do not forget &acute;!'                    |
       | Unclosed hr          | 'Let us add a thematic <hr> break.'                                                  |
+
+  @javascript
+  Scenario Outline: Request translation of a page with HTML5 and Custom Tags into French.
```
Comment (cyberwolf) [inline tests/features/tmgmt/tmgmt_poetry.feature]: What do you mean with "Custom" tags?

File: profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module [hunk 4]
```diff
@@ -1657,3 +1656,54 @@ function multisite_drupal_toolbox_features_post_restore($op, $items) {
     }
   }
 }
+
+/**
+ * Remove PHP Warning logs from HTML5 during a loadHTML.
+ *
+ * Extend SimpleXMLElement function.
```
Comment (cyberwolf) [inline profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module]: It rather enhances the behaviour of DOMDocument::LoadHTML(), not SimpleXMLElement.

File: profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module [hunk 5]
```diff
@@ -1657,3 +1656,54 @@ function multisite_drupal_toolbox_features_post_restore($op, $items) {
...
+  $errors = libxml_get_errors();
+  libxml_use_internal_errors($old);
+  multisite_drupal_toolbox_filter_errors($errors);
+  return TRUE;
+}
+
+/**
+ * Remove PHP Warning logs from HTML5 during a loadHTMLFile.
+ *
+ * Extend SimpleXMLElement function.
```
Comment (cyberwolf) [inline profiles/common/modules/custom/multisite_drupal_toolbox/multisite_drupal_toolbox.module]: It rather enhances the behaviour of DOMDocument::LoadHTML(), not SimpleXMLElement.

File: tests/features/tmgmt/tmgmt_poetry.feature [hunk 6]
```diff
@@ -240,3 +240,34 @@ Feature: TMGMT Poetry features
...
+    And I press "Apply"
+    Then I click "Translate" in the "primary_tabs" region
+    And I select the radio button "" with the id "edit-languages-fr"
+    And I press "Request translation"
+    And I wait
+    And I press "Submit to translator"
+    And I store the job reference of the translation request page
+    Then the poetry translation service received the translation request
+    And the translation request has version 0
+    Then I go to "admin/poetry_mock/dashboard"
```
Comment (cyberwolf) [inline tests/features/tmgmt/tmgmt_poetry.feature]: Then should be used for expected outcomes only. Use When instead here.

File: tests/features/tmgmt/tmgmt_poetry.feature [hunk 7]
```diff
@@ -240,3 +240,34 @@ Feature: TMGMT Poetry features
...
+  Scenario Outline: Request translation of a page with HTML5 and Custom Tags into French.
+    Given I am logged in as a user with the 'administrator' role
+    And I go to "node/add/page"
+    And I select "Basic HTML" from "Text format"
+    And I fill in "Title" with "<title>"
+    And I fill in "Body" with "<body>"
+    And I press "Save"
+    And I select "Published" from "state"
+    And I press "Apply"
+    Then I click "Translate" in the "primary_tabs" region
```
Comment (cyberwolf) [inline tests/features/tmgmt/tmgmt_poetry.feature]: Then should be used for expected outcomes only. Use When or And instead here.

**Comment by cyberwolf** [comment]
👍

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry_mock/src/Mock/PoetryMock.php [hunk 8]
```diff
@@ -268,11 +268,10 @@ private static function translateRequestContent($content, $language) {
     $xml_content = simplexml_load_string($decoded_content);
     // Add language prefix to the title and body first paragraph.
     $title = (string) $xml_content->body->div->div[0];
-    $body = (string) $xml_content->body->div->div[1];
     // Overwriting title with language prefix.
     $xml_content->body->div->div[0] = "[$language] " . $title;
-    // Overwriting body with language prefix.
-    $xml_content->body->div->div[1] = '<p>' . "[$language] " . $body . '</p>';
```
Comment (netlooker) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry_mock/src/Mock/PoetryMock.php]: By this we are losing body content. But indeed it is not something important and can cause other issues.

File: tests/features/tmgmt/tmgmt_poetry.feature [hunk 9]
```diff
@@ -240,3 +240,34 @@ Feature: TMGMT Poetry features
...
+    Given I am logged in as a user with the 'administrator' role
+    And I go to "node/add/page"
+    And I select "Basic HTML" from "Text format"
+    And I fill in "Title" with "<title>"
+    And I fill in "Body" with "<body>"
+    And I press "Save"
+    And I select "Published" from "state"
+    And I press "Apply"
+    When I click "Translate" in the "primary_tabs" region
+    And I select the radio button "" with the id "edit-languages-fr"
```
Comment (netlooker) [inline tests/features/tmgmt/tmgmt_poetry.feature]: This step will change after simplification of Poetry UI - NEXTEUROPA-7719

**Comment by netlooker** [comment]
👍

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry_mock/src/Mock/PoetryMock.php (hunk 8 above)
Comment (voidtek) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry_mock/src/Mock/PoetryMock.php]: We are not losing body content, 

+    // Adding language prefix into the body.
 +    $xml_content->body->div->div[1]->p[] = "[$language]";

We add an additional tag "p" with the language into the body.

**Pull request merged**